IdentityServer/v6/docs/content/fundamentals/keys/automatic_key_management.md

```diff
@@ -0,0 +1,133 @@
+property on the [*IdentityServerOptions*]({{< ref
+"/reference/options#key-management" >}}). 
+
+### Managed Key Lifecycle
+Keys created by Automatic Key Management move through several phases. First, new
+keys are announced, that is, they are added to the list of keys in discovery,
+but not yet used for signing. After a configurable amount of *PropagationTime*,
+keys are promoted to be signing credentials, and will be used by IdentityServer
+to sign tokens. Eventually, enough time will pass that the key is older than the
+configurable *RotationTime*, at which point the key is retired, but kept in
+discovery for a configurable *RetentionDuration*. After the *RetentionDuration*
+has passed, keys are removed from discovery, and optionally deleted.
+
+The default is to rotate keys every 90 days, announce new keys with 14 days of
+propagation time, retain old keys for a duration of 14 days, and to delete keys
+when they are retired. All of these options are configurable in the
+*KeyManagement* options. For example:
+
+```cs
+var builder = services.AddIdentityServer(options =>
+{   
+    // new key every 30 days
+    options.KeyManagement.RotationInterval = TimeSpan.FromDays(30);
+    
+    // announce new key 2 days in advance in discovery
+    options.KeyManagement.PropagationTime = TimeSpan.FromDays(2);
+    
+    // keep old key for 7 days in discovery for validation of tokens
+    options.KeyManagement.RetentionDuration = TimeSpan.FromDays(7);
+
+    // don't delete keys after their retention period is over
+    options.KeyManagement.DeleteRetiredKeys = false;
+});
+```
+
+### Key storage
+Automatic Key Management stores keys through the abstraction of the
+[ISigningKeyStore]({{<ref "/data/operational/keys">}}). You can implement this
+extensibility point to customize the storage of your keys (perhaps using a key
+vault of some kind), or use one of the two implementations of the
+*ISigningKeyStore* that we provide:
+ - the default *FileSystemKeyStore*, which writes keys to the file system.
+ - the [EntityFramework operational store]({{<ref "/data/ef#operational-store">}}) which writes keys to a database using
+   EntityFramework.
+
+The default *FileSystemKeyStore* writes keys to the *KeyPath* directory
+configured in your IdentityServer host, which defaults to the directory
+*~/keys*. This directory should be excluded from source control. 
+
+If you are deploying in a load balanced environment and wish to use the
+*FileSystemKeyStore*, all instances of IdentityServer will need read/write
+access to the *KeyPath*.
+
+```cs
+var builder = services.AddIdentityServer(options =>
+{   
+    // set path to store keys
+    options.KeyManagement.KeyPath = "/home/shared/keys";
+});
+```
+
+### Encryption of Keys at Rest
+The keys created by Automatic Key Management are sensitive cryptographic secrets
+that should be encrypted at rest. By default, keys managed by Automatic Key
+Management are protected at rest using ASP.NET Core Data Protection. This is
```

Comment:
Maybe a link on "Data Protection" to somewhere useful (prolly in our docs so they know they need to configure it)?